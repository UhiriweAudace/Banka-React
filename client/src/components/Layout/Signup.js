import React, { Component } from 'react'

class Signup extends Component {

  render() {
    return (
      <div className="container">
        <main>
          <div className="main-content">
            <div className="side-1"></div>
            <div className="side-2">
              <div className="form">
                <h3 className="d-purple">Signup</h3>          
                <hr />
                <div className="rows">
                  <div className="col-1">
                    <label>Firstname <span className="is-required">*</span></label><br/>
                    <input type="text" name="firstname" id="firstname" placeholder="Type your firstname" /><br/>
                    <span id="error-message" className="error">Firstname field is required</span>
                  </div>
                  <div className="col-1">
                    <label>Lastname <span className="is-required">*</span></label><br/>
                    <input type="text" name="lastname" id="lastname" placeholder="Type your lastname" /><br/>
                    <span id="error-message" className="error">Lastname field is required</span>
                  </div>
                </div>
                
                <div className="rows">
                  <div className="col-1">
                    <label>Email <span className="is-required">*</span></label><br/>
                    <input type="text" name="email" id="email" placeholder="Type your e-mail" /><br/>
                    <span id="error-message" className="error">Email field is required</span>
                  </div>
                  <div className="col-1">
                    <label>Password <span className="is-required">*</span></label><br/>
                    <input type="password" name="password" id="password" placeholder="Type your Password" /><br/>
                    <span id="error-message" className="error">Password field is required</span>
                  </div>
                </div>
                <div className="rows">
                  <div className="col-1">
                    <label>Phone Number <span className="is-required"></span></label><br/>
                    <input type="number" name="phone" id="phone" placeholder="Type your Phone Number" /><br/>
                    <span id="error-message" className="error"></span>
                  </div>
                  <div className="col-1">
                    <label>Address <span className="is-required"></span></label><br/>
                    <input type="text" name="address" id="address" placeholder="Type your current address city" /><br/>
                    <span id="error-message" className="error"></span>
                  </div>
                </div>
                <div className="">
                  <div className="col-1">
                    <button className="btn-1 primary" id="register">Register</button>
                  </div>

                </div>
              </div>
            </div>
            <div className="side-3"></div>
          </div>
        </main>
      </div>       
    )
  }
}

export default Signup
