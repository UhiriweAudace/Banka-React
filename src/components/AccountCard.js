import React from 'react'

const AccountCard = ({account}) => {
  return (
    <div className="t-row" key={account.accountNumber}>
      <label htmlFor="" className={`white ${account.type === 'saving' ? 'lbl-debit' : 'lbl-credit'}`}>{account.type === 'saving' ? 'Saving' :'Current'} Account</label>
      <div className="t-details white">
        <label htmlFor="" className="small"> N<u>o</u> : 00040-06565-10236-{account.accountNumber}</label>
        <label htmlFor="" className="small">currently on this accounts, You have {account.balance} Rwf.</label>
      </div>
      <div className="t-buttons">
        <button className="btn-flat primary white" id="details" data-set={account.accountNumber}>Details</button>
      </div>
    </div>
  );
}

export default AccountCard
