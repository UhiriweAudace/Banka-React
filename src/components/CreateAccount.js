import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';
import { createAccount } from '../redux/actions/createAccount.action';

import '../assets/css/style.css';
import '../assets/css/signup.css';
import '../assets/css/client.css';

class CreateAccount extends Component {
  state = {
    nationalId: '',
    countryResidence: '',
    DateOfBirth: '',
    countryBirth: '',
    currency: '',
    phone: '',
    address: '',
    accountType: 'current'
  };

  onchangeHandler = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  onSubmitHandler = e => {
    e.preventDefault();
    const { accountType, currency, DateOfBirth, nationalId } = this.state;

    if (accountType === '') return toast.warn('Account Type is required!');
    if (currency === '') return toast.warn('Currency is required!');
    if (DateOfBirth === '') return toast.warn('Date Of Birth is required!');
    if (nationalId === '') return toast.warn('National Id is required!');
    const accountData = {
      type: accountType
    };
    this.props.createAccount(accountData);
  };
  componentDidMount() {
    if (!this.props.login.isAuthenticated) {
      window.location.href = '/auth/signin';
    }
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.login.isAuthenticated) {
      this.props.history.push('/accounts/new');
    } else {
      window.location.href = '/auth/signin';
    }

  }

  render() {
    const { account } = this.props;
    if (account !== undefined && account.status === 201) {
      toast.success('   Account created Successfully!');
    }
    console.log('User account type in render::::', this.props);

    return (
      <div className='main create-account' id='create-account'>
        <div></div>
        <div>
          <div className='form'>
            <h3 className='d-purple'>Create New Bank Account</h3>
            <hr />
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  National Identity Card Number{' '}
                  <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='number'
                  name='nationalId'
                  id='id'
                  placeholder='Type your ID Number'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  id field is required
                </span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Country Residence <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='text'
                  name='countryResidence'
                  id='country-residence'
                  placeholder='Type your Country Residence'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  Lastname field is required
                </span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Country Of Birth <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='text'
                  name='countryBirth'
                  id='countrybirth'
                  placeholder='Type your Country of Birth'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'></span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Date of Birth <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='date'
                  name='DateOfBirth'
                  id='DateOfBirth'
                  placeholder='Type your Date Of Birth'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  DateOfBirth field is required
                </span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Currency Type <i>(eg: For Rwandan Francs => Rwf)</i>{' '}
                  <span className='is-required'>*</span>
                </label>
                <br />
                <input
                  type='text'
                  name='currency'
                  id='currency'
                  placeholder='Type your currency type'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'>
                  Currency field is required
                </span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Phone Number <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='number'
                  name='phone'
                  id='phone'
                  placeholder='Type your Phone Number'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error'></span>
              </div>
            </div>
            <div className='rows'>
              <div className='col-1'>
                <label htmlFor=''>
                  Address <span className='is-required'></span>
                </label>
                <br />
                <input
                  type='text'
                  name='address'
                  id='address'
                  placeholder='Type your current address city'
                  onChange={this.onchangeHandler}
                />
                <br />
                <span id='error-message' className='error-message'></span>
              </div>
              <div className='col-1'>
                <label htmlFor=''>
                  Account Type <span className='is-required'>*</span>
                </label>
                <br />
                <select
                  name='accountType'
                  id='account-type'
                  onChange={this.onchangeHandler}
                >
                  <option value='0' disabled>
                    Select Account type
                  </option>
                  <option value='current'>Current Account</option>
                  <option value='saving'>Saving Account</option>
                </select>
                <br />
                <span id='error-message' className='error-message'></span>
              </div>
            </div>
            <div className=''>
              <div className='col-1'>
                <button
                  className='btn-1 primary'
                  id='create'
                  type='submit'
                  onClick={this.onSubmitHandler}
                >
                  Create
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
CreateAccount.propTypes = {
  login: PropTypes.object.isRequired,
  createAccount: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  login: state.login,
  account: state.accountData
});

export default connect(
  mapStateToProps,
  { createAccount }
)(CreateAccount);
