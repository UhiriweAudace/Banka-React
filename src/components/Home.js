import React, { Component } from 'react'; 
import { Link} from 'react-router-dom';
import '../assets/css/style.css';

class Home extends Component {
  render() {
    return (
      <div>
        <div class="main-content">
          <div class="side-1"></div>
          <div class="side-2">
            <div class="center top">
              <h1 class="heading-1 white">Banka</h1>
              <p class="p-1 white">Your Bank Account In Your Hands.</p>
              <p class="p-1 white">Please Visit Our Nearest Branch For Deposit and Withdrawal Operations!</p>
              <Link to='/auth/signin'>
                <button class="btn-1 primary" id="started">
                  Get Started
                </button>
              </Link>
            </div>
          </div>
          <div class="side-3"></div>
        </div>
      </div>
    )
  }
}

export default Home
