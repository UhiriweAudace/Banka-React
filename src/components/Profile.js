import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from "react-router-dom";
import { capitalize } from '../utils/capitilize';
class Profile extends Component {
  componentDidMount() {
    if (this.props.login.isAuthenticated) {
      this.props.history.push('/profile');
    } else {
      window.location.href = '/';
    }
  }
  render() {
    console.log('::::::::::::::::>>>', this.props.login);
    const { login } = this.props;
    return (
      <div>
        <div className='main prof' id='prof'>
          <div></div>
          <div className='profile secondary'>
            <div className='profile-card'>
              <div className='profile-left'>
                <div className='profile-image center'>
                  {/* <!-- <h5 className="heading-5">Your Profile Picture</h5> --> */}
                </div>
                <button className='btn btn-upload'>
                  Upload Your Profile Picture
                </button>
              </div>
              <div className='profile-content'>
                <h3 className='p-bottom'>Personal Details</h3>
                <ul className='no-bullet '>
                  <li className='secondary'>
                    {capitalize(login.user.firstname)} {''}{' '}
                    {capitalize(login.user.lastname)}
                  </li>
                  <li className='secondary'>{login.user.email}</li>
                  <li className='secondary'>(+250) 785166463</li>
                  <li className='secondary'>Kigali, Rwanda</li>
                  <li className='secondary'>#Current Balance 0 Rwf</li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login
});
export default connect(mapStateToProps)(Profile);
