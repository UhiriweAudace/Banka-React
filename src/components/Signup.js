import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';

import { signup } from '../redux/actions/signup.action';

export class Signup extends Component {
  state = {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    address: ''
  };

  componentDidMount(){
    if (this.props.login !== undefined) {
      if (this.props.login.isAuthenticated) {
        this.props.history.push('/profile')
      } 
    }
  }
  componentWillReceiveProps(nextProps){
    if (this.props.login !== undefined) {
      if (nextProps.login.isAuthenticated) {
        this.props.history.push('/profile')
      }
    }
    if (
      nextProps.signupUser.errors !== undefined &&
      nextProps.signupUser.errors.data !== undefined &&
      !nextProps.signupUser.isAuthenticated

    ) {
      return toast.warn(
        `${nextProps.signupUser.errors.data.error ||
          nextProps.signupUser.errors.data.message}`
      );
    }
  }

  onChangeHandler = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };
  onSubmitHandler = e => {
    e.preventDefault();
    const { firstname, lastname, email, password } = this.state;
    const data = {
      firstname,
      lastname,
      email,
      password
    };
    this.props.signup(data);
  };
  render() {    
    return (
      <div>
        <div className='main-content'>
          <div className='side-1'></div>
          <div className='side-2'>
            <div className='form'>
              <h3 className='d-purple'>Signup</h3>
              <hr />
              <div className='rows'>
                <div className='col-1'>
                  <label htmlFor=''>
                    Firstname <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='text'
                    name='firstname'
                    id='firstname'
                    placeholder='Type your firstname'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Firstname field is required
                  </span>
                </div>
                <div className='col-1'>
                  <label htmlFor=''>
                    Lastname <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='text'
                    name='lastname'
                    id='lastname'
                    placeholder='Type your lastname'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Lastname field is required
                  </span>
                </div>
              </div>
              <div className='rows'>
                <div className='col-1'>
                  <label htmlFor=''>
                    Email <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='text'
                    name='email'
                    id='email'
                    placeholder='Type your e-mail'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Email field is required
                  </span>
                </div>
                <div className='col-1'>
                  <label htmlFor=''>
                    Password <span className='is-required'>*</span>
                  </label>
                  <br />
                  <input
                    type='password'
                    name='password'
                    id='password'
                    placeholder='Type your Password'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'>
                    Password field is required
                  </span>
                </div>
              </div>
              <div className='rows'>
                <div className='col-1'>
                  <label htmlFor=''>
                    Address <span className='is-required'></span>
                  </label>
                  <br />
                  <input
                    type='text'
                    name='address'
                    id='address'
                    placeholder='Type your current address city'
                    onChange={this.onChangeHandler}
                  />
                  <br />
                  <span id='error-message' className='error'></span>
                </div>
              </div>
              <div className=''>
                <div className='col-1'>
                  <button
                    className='btn-1 primary'
                    id='register'
                    type='submit'
                    onClick={this.onSubmitHandler}
                  >
                    Register
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className='side-3'></div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  signupUser: state.signup
});
export default connect(
  mapStateToProps,
  { signup }
)(Signup);
