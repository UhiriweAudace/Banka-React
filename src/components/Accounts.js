import React, { Component } from 'react';
import { connect } from 'react-redux';
import AccountCard from './AccountCard';
import {toast} from 'react-toastify';
import {getAllAccounts} from '../redux/actions/account.action';

class Accounts extends Component {

  componentDidMount() {
    if (!this.props.login.isAuthenticated) {
      window.location.href = '/auth/signin';
    }
    this.props.getAllAccounts();
  }
  render() {
    const { accounts} = this.props;
    console.log('User account type in render::::', accounts.accounts);
    return (
      <div className="main my-bank-accounts" id="my-bank-accounts">
        <div></div>
        <div className="transaction-list">
          {
            accounts !== undefined && accounts.accounts.data !== undefined ? accounts.accounts.data.map( account => {
              toast.success(`    ${accounts.accounts.message}`)
              return (
                <AccountCard account={account}/>
              );
            }) : (
              <div className="t-row">
                <label htmlFor="" className='white lbl-debit'></label>
                <div className="t-details white">
                  <label htmlFor="" className="small"> </label>
                  <label htmlFor="" className="small">  Whoops! Currently, You don't have any Bank Accounts.</label>
                </div>
                <div className="t-buttons">
                  {/* <button className="btn-flat primary white" id="details" >Details</button> */}
                </div>
              </div>
            )
          }
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  login: state.login,
  accounts: state.account,
})
export default connect(mapStateToProps, { getAllAccounts })(Accounts);
