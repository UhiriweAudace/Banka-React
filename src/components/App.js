import React, { Component } from 'react';
import { BrowserRouter, Route, Switch} from 'react-router-dom';
import Header from './Layout/Header';
import Home from './Home';
import Signin from './Signin';
import Signup from './Signup';


class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Header/>
          <Switch>
            <Route exact path='/' component={Home} />
            <Route exact path='/auth/signin' component={Signin} />
            <Route exact path='/auth/signup' component={Signup} />
          </Switch>
        </div>
      </BrowserRouter>
    )
  }
}

export default App
