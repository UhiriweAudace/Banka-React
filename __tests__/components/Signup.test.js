import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Signup } from '../../src/components/Signup';

let wrapper;
describe('Signup Page', () => {
  test('It should render the Signup Page', () => {
    const props = {
      signup: {
        errors: { message: 'Signup successfully' }
      },
      history: {
        push: jest.fn()
      },
      signup: jest.fn()
    };
    wrapper = shallow(<Signup {...props} />);
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  test('It should allow users to enter his/her information', () => {
    const email = wrapper.find('#email');
    const password = wrapper.find('#password');
    const firstName = wrapper.find('#firstname');
    const lastName = wrapper.find('#lastname');
    const address = wrapper.find('#address');
    
    email.simulate('change', {
      target: { value: 'u.audace@gmail.com', name: 'email' }
    });
    password.simulate('change', {
      target: { value: 'Uhiriwe@2222', name: 'password' }
    });
    firstName.simulate('change', {
      target: { value: 'Audace', name: 'firstname' }
    });
    lastName.simulate('change', {
      target: { value: 'Uhiriwe', name: 'lastname' }
    });
    address.simulate('change', {
      target: { value: 'Kigali', name: 'address' }
    });
    expect(wrapper.state('email')).toEqual('u.audace@gmail.com');
    expect(wrapper.state('password')).toEqual('Uhiriwe@2222');
    expect(wrapper.state('firstname')).toEqual('Audace');
    expect(wrapper.state('lastname')).toEqual('Uhiriwe');
    expect(wrapper.state('address')).toEqual('Kigali');
  });
});
describe('submit button test...', () => {
  test('User will be able to click on submit button and send an action', () => {
    let submitButton;
    const preventDefault = jest.fn();
    submitButton = wrapper.find('#register');
    submitButton.simulate('click', { preventDefault });
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });
});
