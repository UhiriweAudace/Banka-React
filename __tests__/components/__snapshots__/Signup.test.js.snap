// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page It should render the Signup Page 1`] = `ShallowWrapper {}`;

exports[`submit button test... User will be able to click on submit button and send an action 1`] = `
<div>
  <div
    className="main-content"
  >
    <div
      className="side-1"
    />
    <div
      className="side-2"
    >
      <div
        className="form"
      >
        <h3
          className="d-purple"
        >
          Signup
        </h3>
        <hr />
        <div
          className="rows"
        >
          <div
            className="col-1"
          >
            <label
              htmlFor=""
            >
              Firstname 
              <span
                className="is-required"
              >
                *
              </span>
            </label>
            <br />
            <input
              id="firstname"
              name="firstname"
              onChange={[Function]}
              placeholder="Type your firstname"
              type="text"
            />
            <br />
            <span
              className="error"
              id="error-message"
            >
              Firstname field is required
            </span>
          </div>
          <div
            className="col-1"
          >
            <label
              htmlFor=""
            >
              Lastname 
              <span
                className="is-required"
              >
                *
              </span>
            </label>
            <br />
            <input
              id="lastname"
              name="lastname"
              onChange={[Function]}
              placeholder="Type your lastname"
              type="text"
            />
            <br />
            <span
              className="error"
              id="error-message"
            >
              Lastname field is required
            </span>
          </div>
        </div>
        <div
          className="rows"
        >
          <div
            className="col-1"
          >
            <label
              htmlFor=""
            >
              Email 
              <span
                className="is-required"
              >
                *
              </span>
            </label>
            <br />
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Type your e-mail"
              type="text"
            />
            <br />
            <span
              className="error"
              id="error-message"
            >
              Email field is required
            </span>
          </div>
          <div
            className="col-1"
          >
            <label
              htmlFor=""
            >
              Password 
              <span
                className="is-required"
              >
                *
              </span>
            </label>
            <br />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Type your Password"
              type="password"
            />
            <br />
            <span
              className="error"
              id="error-message"
            >
              Password field is required
            </span>
          </div>
        </div>
        <div
          className="rows"
        >
          <div
            className="col-1"
          >
            <label
              htmlFor=""
            >
              Address 
              <span
                className="is-required"
              />
            </label>
            <br />
            <input
              id="address"
              name="address"
              onChange={[Function]}
              placeholder="Type your current address city"
              type="text"
            />
            <br />
            <span
              className="error"
              id="error-message"
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="col-1"
          >
            <button
              className="btn-1 primary"
              id="register"
              onClick={[Function]}
              type="submit"
            >
              Register
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="side-3"
    />
  </div>
</div>
`;
